File: luasrc/model/cbi/v2ray/reverse.lua
-- Copyright 2019 Xingwang Liao <kuoruan@gmail.com>
-- Licensed to the public under the MIT License.

local dsp = require "luci.dispatcher"

local m, s, o

m = Map("v2ray", "%s - %s" % { translate("V2Ray"), translate("Reverse") },
	translatef("Details: %s", "<a href=\"https://www.v2ray.com/en/configuration/reverse.html#reverseobject\" target=\"_blank\">ReverseObject</a>"))

s = m:section(NamedSection, "main_reverse", "reverse")
s.anonymous = true
s.addremove = false

o = s:option(Flag, "enabled", translate("Enabled"))
o.rmempty = false

o = s:option(DynamicList, "bridges", translate("Bridges"),
	translatef("A list of bridges, format: <code>tag|domain</code>. eg: %s", "bridge|test.v2ray.com"))

o = s:option(DynamicList, "portals", translate("Portals"),
	translatef("A list of portals, format: <code>tag|domain</code>. eg: %s", "portal|test.v2ray.com"))

return m

File: htdocs/luci-static/resources/view/v2ray/reverse.js
/**
 * @license
 * Copyright 2020 Xingwang Liao <kuoruan@gmail.com>
 *
 * Licensed to the public under the MIT License.
 */
"use strict";"require form";return L.view.extend({render:function(){var e=new form.Map("v2ray","%s - %s".format(_("V2Ray"),_("Reverse")),_("Details: %s").format('<a href="https://www.v2ray.com/en/configuration/reverse.html#reverseobject" target="_blank">ReverseObject</a>')),r=e.section(form.NamedSection,"main_reverse","reverse");return r.addremove=!1,r.option(form.Flag,"enabled",_("Enabled")).rmempty=!1,r.option(form.DynamicList,"bridges",_("Bridges"),_("A list of bridges, format: <code>tag|domain</code>. eg: %s").format("bridge|test.v2ray.com")),r.option(form.DynamicList,"portals",_("Portals"),_("A list of portals, format: <code>tag|domain</code>. eg: %s").format("portal|test.v2ray.com")),e.render()}});